packages/mdc-chips/chip/foundation.js

```diff
@@ -175,7 +194,10 @@ class MDCChipFoundation extends MDCFoundation {
     evt.stopPropagation();
     if (evt.type === 'click' || evt.key === 'Enter' || evt.keyCode === 13) {
       this.adapter_.notifyTrailingIconInteraction();
-      this.adapter_.addClass(cssClasses.CHIP_EXIT);
+      if (this.shouldRemoveOnTrailingIconClick_) {
+        // Begins the exit animation which leads to removal of the chip
+        this.adapter_.addClass(cssClasses.CHIP_EXIT);
```

Comment:
It occurs to me that this is effectively doing the same thing that `beginExit` on the component is doing. Do you think we should create a `beginExit` foundation method and proxy to it from the component's method, so that both can reuse the same code?

Reply:
I thought about that but realized that's actually more code?
...But I guess it makes sense pattern-wise. Ok I'll update.

Reply:
It probably is more code ultimately, but in this case the trade-off is avoiding repeating the same logic across modules which risks skew later.